src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -226,15 +226,15 @@ private void UpdateStartingStreams(DateTimeOffset now)
             // Don't create Encoder and Decoder as they aren't used now.
             Exception? error = null;
 
-            // TODO should we await the control stream task?
-            var controlTask = CreateControlStream(application);
+            // Don't delay setting up the connection on the control stream being ready.
+            _ = CreateControlStream(application).Preserve();
```

Comment:
I liked that the old comment was at least admitting that not awaiting a task at all is probably not ideal. Is it okay if ProcessRequestsAsync exits before CreateControlStream does? What happens if Http3Connection.Abort() is called after middleware completes? Is that ever a possibility in reality?

I think we should probably just avoid these questions and await this ValueTask in a finally or something. It will never throw because of the `catch (Exception ex)`.